lib/wallet.js

```diff
@@ -56,18 +65,47 @@ module.exports = function(options) {
         // Compute the hash of the public generator with the sub-sequence number
         sec = sjcl.bn.fromBits(firstHalfOfSHA512(append_int(append_int(publicGenerator.toBytesCompressed(), 0), i)));
         i++;
-        // If the hash is equal to or greater than the SECp256k1 order, increment the sequenceuence and retry
+        // If the hash is equal to or greater than the SECp256k1 order, increment the sequence and retry
       } while (!sjcl.ecc.curves.c256.r.greaterEquals(sec));
       // Treating this hash as a private key, compute the corresponding public key as an EC point. 
       return sjcl.ecc.curves.c256.G.mult(sec).toJac().add(publicGenerator).toAffine();
     },
 
     getAddress: function(){
-      var privateKey = this.getPrivateKey(this.secret);
-      var privateGenerator = this.getPrivateGenerator(privateKey);
-      var publicGenerator = PublicGenerator.fromPrivateGenerator(privateGenerator).value;
+      var publicGenerator = this.getPublicGenerator().value;
+      var publicKey = this.getPublicKey(publicGenerator);
+      return RippleAddress.fromPublicKey(publicKey);
+    },
+
+    getAddressFromValidationPublicKey: function(validation_public_key){
+      var compressedPublicGenerator = base58.decode_base_check(28, validation_public_key);
+      var ec_x = compressedPublicGenerator.slice(1);
+      var point = ec.curve.pointFromX(ec_x, compressedPublicGenerator[0] === 0x03)
+      var x = sjcl.bn.fromBits(sjcl.codec.bytes.toBits(ec_x));
+      var y = sjcl.bn.fromBits(sjcl.codec.bytes.toBits(point.y.toArray()))
+      var publicGenerator = new sjcl.ecc.point (sjcl.ecc.curves.c256, x, y)
```

Comment:
aye :)

Oh, I just found this: https://github.com/sublimator/ripple-data-sign/blob/master/pub_key.coffee#L170-L190